### ISSUE TITLE
Secure Connection between GPS Client and Host

### ISSUE THREAD

**Issue by dschense** [tananaev/traccar#1942]
Connection to the Webinterface could be secured over reverse Proxy with https. This is great and works like a charme.
But is it also possible to secure the Connection between the GPS Client (for me its a Android Mobile Device running the Traccar APP and connection over TCP Port 5055) and the Host running the Traccar instance?

The Connection looks like this:

```
2016-05-12 11:47:38  INFO: [36330907] connected
2016-05-12 11:47:39 DEBUG: [36330907: 5055 < #IPOfTheDevice] HEX: 
(This is not real. Just a example)
345634634gkdgekgegkegk34t4g34gß4kgß34kg3k3ßfk3ßfk3ßfk3ßfk3ßfk3ßtk3ßtk3ßk3ßtk3ßtk3dßkt3ßkdt3ßtk3ßtkd3ßtkd3ßtk3ßtk3ßtdk3ßtkg3ßtk3ßtk23tk34ß3ß6zj3ß6j3ß5j2ßd24k2´05k2´05kv2´5kv2´5kv2´5kv235kv34ßk6vb3ßzkvb3ßvb6k3ß6kv3ß6kvb3ß6kvb34ßkv3ßkct3ktß3kd3kdt3ükodäseoküghkwezwpjwzeüopzjwpjzw4jüjtzw3ßktd3ßk4dtwüßküktwpeojwrtwpejtzwjtfpüwjdpüwükdwoptfkweüzkweüzkowüozkwerjwwfeprjweijrzwüiejzüejzüwoezw+oekrzwerkfzwkozw+eokzw+zk+0zk+kzwk0ww304kzw30ktzw30ktzw3#tkw3tzk
2016-05-12 11:47:39 DEBUG: [36330907: 5055 > #IPOfTheDevice] HEX: 
(This is not real. Just a example)
345634634gkdgekgegkegk34t4g34gß4kgß34kg3k3ßfk3
2016-05-12 11:47:39  INFO: [36330907] disconnected
2016-05-12 11:47:22  INFO: [C9825705] id: 9, time: 2016-05-12 10:56:53, lat: 22.69108, lon: 27.67546, speed: 1.7, course: 22.1
```
It sends some Stuff to Host:5055

**Comment by tananaev** [comment]
Traccar Client doesn't support secure connection at the moment, but it's certainly possible to add support for it. The problem is that it won't help with spoofing.

**Comment by dschense** [comment]
Do you have any Idea how to secure the System against Spoofing.

**Comment by tananaev** [comment]
There has to be some authentication. For example, the simplest option is a password.

**Comment by dschense** [comment]
Hm, okay. but this has to be implemented to the Server and to the Client. Its not possible at the Moment to secure this with a "dirty-Hack" or something like this.
Seems to be a bigger project to add this feature, isnt it ?

**Comment by tananaev** [comment]
Yes, it has to be implemented on the both client and server side. I can't think of any simple workaround.

**Comment by dschense** [comment]
Okay, hmmm.. then i think this should be a first priority Feature to be implemented. Safety first.. in my opinion there is a not that small risk without the security feature.

**Comment by tananaev** [comment]
I don't think it's going to be first priority feature, but I'll consider it for one of the next releases.

**Comment by dschense** [comment]
Okay .. maybe someone can help with creating a modded osmand protocol to accept https connection between Client (Android) and Server. Looked inside the code, but my Java Skills are not that good to do it myself.

**Comment by tananaev** [comment]
Send me an email (address is in profile).

**Comment by dschense** [comment]
@tananaev you got a Mail.

**Comment by FX7** [comment]
I would agree with @dschense, that it would be wonderfull to secure the client/host connection.

Without any security in the connection, not only spoofing is possible but also anybody could read my position data.

I really would like to help you a bit with this. I would say, I know enough about java to do the coding. But I don't know much about maven or anythink about android programming. Also I don't have to much time ... Long story short: With some help, i could get this running!

**Comment by dschense** [comment]
For Client - Server Connection with Android OsmAnd Protocol is used:

On the server side this could be the right files to look at:

https://github.com/tananaev/traccar/blob/master/src/org/traccar/protocol/OsmAndProtocol.java
https://github.com/tananaev/traccar/blob/master/src/org/traccar/protocol/OsmAndProtocolDecoder.java

And on the client side:

https://github.com/tananaev/traccar-client-android/blob/master/app/src/main/java/org/traccar/client/ProtocolFormatter.java

Iam not a Java Pro (at learning phase - beginner rank XD) but I think theses files are the place to start (and maybe end)

**Comment by dschense** [comment]
@FX7 how to get in contact? My Email is in my Profile. (Email or Hangouts)

**Comment by FX7** [comment]
Frontend (Client and Server) is needed. Also some database scheme extension on the server side and encoding on the client side. Hope we can get this to work!

**Comment by dschense** [comment]
#2016 

better to go here
